src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -600,6 +649,7 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
 
                       handler->AddResponse(cachedCommandPath, response);
                       handle.Release();
+                      gOtaSender.ResetState();
```

Comment:
This seems to be assuming that `response` is not a success here.  What guarantees that, exactly?  As far as I can tell, nothing.  We could be landing here with `hasUpdate` true and `isBDXProtocolSupported` false, and sending a "yes, do stuff" to the other side, then clearing our state....

What should probably happen here is that if `hasUpdate` is false we send the response as-is, but if `hasUpdate` is true we create a new response with the right state (because `isBDXProtocolSupported` must be false in that case). and send that.  Then it's OK to reset our state, because we are guaranteeing an error response.

Reply:
The point is, at this point in the code `hasUpdate` can be _true_.

Reply:
yes but protocol supported is false and thats not a good state either. I am going to take a look after my appointment but I think we will send protocol not supported to the requestor and requestor will quit at that point.

Reply:
>  I am going to take a look after my appointment but I think we will send protocol not supported to the requestor

Will we?  Why?  We are going to send _whatever status our delegate set_.  My point is that we should be explicitly sending "protocol not supported" if that's the state we are in.

Reply:
done.